### PR TITLE
Remove --enable-authentication option

### PR DESCRIPTION
**Pull request by Yaliang**
`--enable-authentication` option in `presto-cli` is misleading because this option doesn't state clearly that it's using Kerberos authentication.
Given the `kerberosRemoteServiceName` variable is required for Kerberos based authentication, we can use the present of `--krb5-remote-service-name` value to determine that the client should setup Kerberos. This behavior has already employed in `presto-jdbc`. 
In addition, since the credential should only be sent over the secured connection, we should also enforce the SSL for Kerberos based authentication, which is same as the check in basic/password authentication.

### CODE REVIEW

File: presto-cli/src/test/java/com/facebook/presto/cli/TestTableNameCompleter.java [hunk 1]
```diff
@@ -27,7 +27,6 @@ public void testAutoCompleteWithoutSchema()
     {
         ClientSession session = new ClientOptions().toClientSession();
         QueryRunner runner = createQueryRunner(session);
-
```
Comment (findepi) [inline presto-cli/src/test/java/com/facebook/presto/cli/TestTableNameCompleter.java]: unrelated change

File: presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java [hunk 2]
```diff
@@ -78,10 +77,12 @@ public QueryRunner(
         setupHttpProxy(builder, httpProxy);
         setupBasicAuth(builder, session, user, password);
 
-        if (kerberosEnabled) {
+        if (kerberosRemoteServiceName.isPresent()) {
+            checkArgument(session.getServer().getScheme().equalsIgnoreCase("https"),
```
Comment (findepi) [inline presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java]: please extract adding this check to a separate commit, especially that it requires second thought

File: presto-cli/src/test/java/com/facebook/presto/cli/TestTableNameCompleter.java (hunk 1 above)
Comment (Yaliang) [reply presto-cli/src/test/java/com/facebook/presto/cli/TestTableNameCompleter.java]: sorry, was messed up when cherrying pick from the other branch. Fixed

File: presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java (hunk 2 above)
Comment (Yaliang) [reply presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java]: done

Comment (findepi) [reply presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java]: Oh, i see we already have such a check for basic auth.

**Comment by electrum** [comment]
Merged, thanks!
